pmsoService/product_flow/features/deliverytrip/app/handlers.py

```
@@ -1,3 +1,346 @@
+    if not can_record_return(actor_user):
+        raise PermissionError("Only storage keepers or manager can record returns")
+
+    # Validate inputs (require client timestamp if business rule demands it)
+    validate_return_quantity(command.returned_quantity)
+    validate_return_timestamp(command.returned_quantity, command.return_received_at)
+    ts = command.return_received_at  # requires user to input else raise validator
+
+    # Fetch current item state
+    delivery_item_id = getattr(command, "delivery_item_id", None) or getattr(
+        command, "id"
+    )
```

Ruff (0.13.1)

Do not call `getattr` with a constant attribute value. It is not any safer than normal property access.

Replace `getattr` with attribute access

(B009)